### ISSUE TITLE
Calculations are stuck after 'verdi daemon restart'.

### ISSUE THREAD

**Issue by greschd** [aiidateam/aiida_core#1274]
After re-starting the daemon, calculations which were previously `RUNNING` / `QUEUED` are stuck, i.e. they are not updated even after the calculation finishes.

Circus log:
```
2018-03-13 10:12:05 circus[11879] [INFO] Starting master on pid 11879
2018-03-13 10:12:05 circus[11879] [ERROR] exception Address already in use caught
2018-03-13 10:12:20 circus[11645] [INFO] Got signal SIG_TERM
2018-03-13 10:12:20 circus[11645] [INFO] Arbiter exiting
2018-03-13 10:12:20 circus[11650] [INFO] Stats streamer stopped
2018-03-13 10:12:20 tornado.application[11650] [ERROR] Exception in callback <functools.partial object at 0x7f80d065de68>
Traceback (most recent call last):
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/tornado/ioloop.py", line 605, in _run_callback
    ret = callback()
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/tornado/stack_context.py", line 277, in null_wrapper
    return fn(*args, **kwargs)
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py", line 536, in <lambda>
    self.io_loop.add_callback(lambda : self._handle_events(self.socket, 0))
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py", line 446, in _handle_events
    zmq_events = self.socket.EVENTS
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/zmq/sugar/attrsettr.py", line 45, in __getattr__
    return self._get_attr_opt(upper_key, opt)
  File "/home/greschd/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/zmq/sugar/attrsettr.py", line 49, in _get_attr_opt
    return self.get(opt)
  File "zmq/backend/cython/socket.pyx", line 501, in zmq.backend.cython.socket.Socket.get
  File "zmq/backend/cython/socket.pyx", line 264, in zmq.backend.cython.socket._getsockopt
  File "zmq/backend/cython/checkrc.pxd", line 25, in zmq.backend.cython.checkrc._check_rc
    raise ZMQError(errno)
ZMQError: Socket operation on non-socket
2018-03-13 10:12:20 circus[11650] [INFO] Stats streamer stopped
2018-03-13 10:12:20 circus[11650] [INFO] Stats streamer stopped
2018-03-13 10:12:20 circus[11645] [INFO] aiida-test_profile stopped
2018-03-13 10:12:20 circus[11645] [INFO] circusd-stats stopped
2018-03-13 10:12:36 circus[11941] [INFO] Starting master on pid 11941
2018-03-13 10:12:36 circus[11941] [INFO] Arbiter now waiting for commands
2018-03-13 10:12:36 circus[11941] [INFO] aiida-test_profile started
2018-03-13 10:12:36 circus[11941] [INFO] circusd-stats started
2018-03-13 10:12:37 circus[11946] [INFO] Starting the stats streamer
2018-03-13 10:24:47 circus[11941] [INFO] aiida-test_profile stopped
2018-03-13 10:24:47 circus[11941] [INFO] aiida-test_profile started
```

**Comment by greschd** [comment]
The following traceback (from shutting down `verdi devel run_daemon` with Ctrl-C) seems to suggest that the signal is not correctly captured while updating the scheduler state.
[traceback.txt](https://github.com/aiidateam/aiida_core/files/1830171/traceback.txt)

**Comment by sphuber** [comment]
Fixed by proxy in PR #1304
